src/Fields/Blueprint.php

```diff
@@ -370,11 +374,49 @@ public function toPublishArray()
         ];
     }
 
+    public function afterSave($callback)
+    {
+        $this->afterSaveCallbacks[] = $callback;
+
+        return $this;
+    }
+
+    public function saveQuietly()
+    {
+        $this->withEvents = false;
+
+        return $this->save();
+    }
+
     public function save()
     {
+        $isNew = is_null(Facades\Blueprint::find($this->handle()));
```

Comment:
One thing I've noticed is that `$isNew` seems to always return `null`, meaning the `BlueprintCreated` event is always dispatched.

I'm not quite sure why that's happening - I have a theory it has something to do with Blink in the [`Blueprint::find`](https://github.com/statamic/cms/blob/3.3/src/Fields/BlueprintRepository.php#L36) method (but then again, there's lots of things that could have changed to break it in the last 1.5 years 😆 )

Reply:
I pushed up a fix. That's because when a blueprint has a namespace, you need to include that in the find() call.